File: Pythoncode/26.删除排序数组中的重复项.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        j = 0
        while j + 1 < len(nums):
            if nums[j+1] != nums[j]:
                i += 1
                nums[i] = nums[j+1]
            j += 1
        return i + 1

File: Cppcode/26.删除排序数组中的重复项.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;
        auto head = nums.begin();
        auto tail = head + 1;
        while (tail != nums.end())
        {
            if(*tail != *(tail-1))
            {
                *(++head) = *tail;
            }
            tail++;
        }
        return head - nums.begin() + 1;
    }
};